std/codec.go

```diff
@@ -29,10 +30,12 @@ type Codec struct {
 	// Keep reference to the amino codec to allow backwards compatibility along
 	// with type, and interface registration.
 	amino *codec.Codec
+
+	anyUnpacker types.AnyUnpacker
 }
 
-func NewAppCodec(amino *codec.Codec) *Codec {
-	return &Codec{Marshaler: codec.NewHybridCodec(amino), amino: amino}
+func NewAppCodec(amino *codec.Codec, anyUnpacker types.AnyUnpacker) *Codec {
```

Comment:
I don't understand having the need for an "app codec" anymore. Why can't the codec for evidence along with it's methods be implemented in the `x/evidence` module?

Reply:
It can, we would basically get rid of a bunch of the codec stuff then. It would only be needed to enable Amino JSON on legacy custom queriers/REST endpoints (which themselves will be deprecated by gRPC in the future)

Reply:
We can still support Amino. Create a hybrid codec and pass that to each module's keeper constructor (instead of a single holistic app-level codec).